### PR TITLE
Rework wait and data dump

### PR DESCRIPTION
**Pull request by perrettecl**

### CODE REVIEW

File: test/process.cpp [hunk 1]
```diff
@@ -349,7 +356,22 @@ TEST_CASE("Process with Huge data")
         auto pid     = process.run();
         CHECK(pid.isValid());
         CHECK(pid);
-        CHECK(process.readAllStandardOutput().size());
+        //CHECK(process.readAllStandardOutput().size());
```
Comment (ericclappier-eaton) [inline test/process.cpp]: Why deactivated ?

File: test/process.cpp [hunk 2]
```diff
@@ -330,8 +337,8 @@ TEST_CASE("Launch 2 process at the time with launcher in separeted thread")
     {
```
Comment (ericclappier-eaton) [inline test/process.cpp:351]: Process finish before timeout. It is not what we want to test (same as previous test)

File: fty/process.h [hunk 3]
```diff
@@ -246,67 +271,102 @@ inline Expected<pid_t> Process::run()
...
-        uint32_t numberOfCycles = 0;
+    // Better to use number of cycle to wait, than using std::chrono, which require a syscall to have exact time.
+    uint64_t numberOfCycles = 0;
 
-        // Number of cycle is timeout (ms) / by the cycle duration (ms) and rounded to upper value
-        uint32_t maxNumberOfCycles = static_cast<uint32_t>(timeoutMs) / waitCycleDurationMs;
-        if (static_cast<uint32_t>(timeoutMs) % waitCycleDurationMs > 0) {
-            maxNumberOfCycles++;
+    // Number of cycle is timeout (ms) / by the cycle duration (ms) and rounded to upper value
+    uint64_t maxNumberOfCycles =timeoutMs / waitCycleDurationMs;
```
Comment (ericclappier-eaton) [inline fty/process.h]: escape after "="

File: fty/process.h [hunk 4]
```diff
@@ -246,67 +271,102 @@ inline Expected<pid_t> Process::run()
...
+                return WSTOPSIG(status);
             }
+            
+            return unexpected("Impossible to identify reason for stop");
         }
     }
 
     // We wait until the end of the program -> wait pid can return for several reason, and we want to get only when our pid returns
-    do {
+    /*do {
```
Comment (ericclappier-eaton) [inline fty/process.h]: Maybe needed if a the child state change (i.e SIGCONT)

**Review by ericclappier-eaton — APPROVED** [review]

File: fty/process.h [hunk 5]
```diff
@@ -47,10 +51,12 @@ class Process
     ~Process();
 
     Expected<pid_t> run();
-    Expected<int>   wait(int timeoutMs = -1, uint32_t waitCycleDurationMs = 100);
 
-    std::string readAllStandardError(int milliseconds = -1);
-    std::string readAllStandardOutput(int milliseconds = -1);
+    //unlimited wait, is limited to uint64_t max value (look reasonable as we univers should be vanish by then)
```
Comment (aquette) [inline fty/process.h]: (looks reasonable as our universe should have vanished by then)

File: fty/process.h (hunk 3 above)
Comment (aquette) [reply fty/process.h]: more a "space"

**Pull request merged**